### PR TITLE
Process Replay: move to pytest

### PR DESCRIPTION
**Pull request by jnewb1**
```TestProcessReplayBase```
base class give a segment and list of processes to replay, and it replays them into ```self.log_msgs``` for analysis by other tests

```TestProcessReplayDiffBase```
base class to perform a diff based on ```self.log_msgs``` and a reference commit

```TestCarProcessReplay```
parameterized to perform a diff on each car's segment, and also checks controlsd was engaged at some point in the routes

parallelization is handled by pytest

all the command line arguments should be the same as before, except they are handled by pytest

there is an additional argument ```--long-diff``` that allows you to see each process's long diff (which formerly was placed into diff.txt)

### CODE REVIEW

**Comment by jnewb1** [comment]
now can run locally with ```--long-diff``` to get the full diff

**Review by thegauravgiri — APPROVED** [review]
Looks good to me

File: selfdrive/test/process_replay/test_car_processes.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  def test_controlsd_engaged(self):
+    if "controlsd" not in self.tested_procs:
+      raise unittest.SkipTest("controlsd was not requested to be tested")
+
+    # check to make sure openpilot is engaged in the route
+    log_msgs = self.log_msgs["controlsd"]
+    self.assertTrue(check_openpilot_enabled(log_msgs), f"Route did not enable at all or for long enough: {self.segment}")
+
+
+if __name__ == '__main__':
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/test_car_processes.py:96]: why is this file renamed?

Comment (jnewb1) [reply selfdrive/test/process_replay/test_car_processes.py:96]: I figured test_processes would make more sense for fuzzy testing or generic process testing, where as these tests are currently more useful for testing car-related stuff

**Review by adeebshihadeh — APPROVED** [review]
The default is slow now. Can you make it multi-threaded by default again?

**Comment by jnewb1** [comment]
> The default is slow now. Can you make it multi-threaded by default again?

with https://github.com/commaai/openpilot/pull/30441 it will now default to threaded (as will all tests)
